BeamlineGeom/inc/StraightSection.hh

```diff
@@ -21,46 +21,51 @@ namespace mu2e {
   public:
 
     StraightSection() : TSSection(),
-      _rIn(0.), _rOut(0.), _halfZ(0.) 
+			_rIn(0.), _rOut(0.), _halfZ(0.), _diffZ(0.)
     {}
 
     StraightSection(double rIn, double rOut, double halfZ, 
                     CLHEP::Hep3Vector const & origin, 
                     CLHEP::HepRotation const & rotation = CLHEP::HepRotation(),
-                    std::string const & materialName = "") :
+                    std::string const & materialName = "", double diffZ=0.) :
       TSSection(origin, rotation, materialName),
-      _rIn(rIn), _rOut(rOut), _halfZ(halfZ)
+      _rIn(rIn), _rOut(rOut), _halfZ(halfZ), _diffZ(diffZ)
     {}
 
     explicit StraightSection( const Tube& tube )  :
       TSSection(tube.originInMu2e(),  tube.rotation(), tube.materialName()),
       _rIn  ( tube.innerRadius() ),
       _rOut ( tube.outerRadius() ),
-      _halfZ( tube.zHalfLength() ) 
+      _halfZ( tube.zHalfLength() ),
+      _diffZ(0.)
     {}
 
     ~StraightSection(){}
 
     void set(double rIn, double rOut, double halfZ, 
              CLHEP::Hep3Vector const & origin, CLHEP::HepRotation const & rotation=CLHEP::HepRotation(),
-             std::string const & materialName = "") {
+             std::string const & materialName = "", double diffZ = 0.) {
       _rIn=rIn;
       _rOut=rOut;
       _halfZ=halfZ;
       _origin=origin;
       _rotation=rotation;
       _materialName=materialName;
+      _diffZ=diffZ;
     }
 
     double rIn() const { return _rIn; }
     double rOut() const { return _rOut; }
     double getHalfLength() const { return _halfZ; }
+    double getLengthDiff() const { return _diffZ; }
+    void   setLengthDiff(double zDiff) { _diffZ=zDiff; }
 
   private:
 
     double _rIn;
     double _rOut;
     double _halfZ;
+    double _diffZ; //amount to end before high Z end
```

Comment:
Slightly clearer comment would be helpful for digging around in the future.  Amount to end of what before high Z end of what?  Not critical for this particular PR, but would be nice to handle soon.

Reply:
Sorry this variable naming/comment aren't very clear! This was to add a configurable buffer between the end of the coil assembly and the upstream/downstream end walls in the TS. This buffer is currently set to 125 mm in: https://github.com/michaelmackenzie/Offline/blob/GeometryUpdates/Mu2eG4/geom/TransportSolenoid_v06.txt#L61-L67